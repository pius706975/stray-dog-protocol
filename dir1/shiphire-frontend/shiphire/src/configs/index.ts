export * from "./Fonts"
export * from "./Colors"
export * from "./LocalStorage"
export * from "./Svg"
export * from "./StatusText"
export * from "./firebase/GoogleSignIn"
export * from "./firebase/AppleSignIn"