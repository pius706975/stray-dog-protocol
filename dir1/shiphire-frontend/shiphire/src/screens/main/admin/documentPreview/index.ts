export {default as AdminDocumentPreview} from './AdminDocumentPreview';
