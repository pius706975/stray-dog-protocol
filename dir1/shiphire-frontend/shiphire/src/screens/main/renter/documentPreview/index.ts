export {default as RenterDocumentPreview} from './RenterDocumentPreview';
