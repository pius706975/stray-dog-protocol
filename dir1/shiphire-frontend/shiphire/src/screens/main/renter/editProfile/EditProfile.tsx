import React from 'react';
import { useTranslation } from 'react-i18next';
import { Button, Text, View } from 'react-native-ui-lib';
import { ScreenLayout } from '../../../../components';
import {
    Color,
    FontFamily,
    FontSize,
    USERDATA,
    getDataFromLocalStorage,
    setDataToLocalStorage,
} from '../../../../configs';
import { TextInput } from 'react-native';
import { addPhoneNumber } from '../../../../hooks';
import { AddPhoneNumberRequest, EditProfileProps } from '../../../../types';
import { useDispatch } from 'react-redux';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';

const EditProfile: React.FC<EditProfileProps> = ({ navigation }) => {
    const { t } = useTranslation('account');
    const [userData, setUserData] = React.useState({
        email: '',
        name: '',
        phoneNumber: '',
        imageUrl: '',
        isVerified: true,
        isCompanySubmitted: true,
    });
    const [renterPhone, setRenterPhone] = React.useState<string>('');
    const mutationAddPhoneNumber = addPhoneNumber();
    const dispatch = useDispatch();
    const { showModal, hideModal } = modalSlice.actions;
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;

    React.useEffect(() => {
        getDataFromLocalStorage(USERDATA).then(res => {
            if (res) {
                setUserData(res);
                setRenterPhone(res.phoneNumber);
            }
        });
    }, []);

    const handleSave = () => {
        dispatch(showProgressIndicator());
        const data: AddPhoneNumberRequest = {
            phoneNumber: renterPhone,
        };
        mutationAddPhoneNumber.mutate(data, {
            onSuccess: () => {
                dispatch(
                    showModal({
                        text: t('EditProfile.successText'),
                        status: 'success',
                    }),
                );
                setTimeout(() => {
                    navigation.reset({
                        index: 0,
                        routes: [{ name: 'MainScreenTab' }],
                    });
                    dispatch(hideProgressIndicator());
                    dispatch(hideModal());
                }, 2000);
                setUserData({
                    ...userData,
                    phoneNumber: renterPhone,
                });
                setDataToLocalStorage(USERDATA, {
                    ...userData,
                    phoneNumber: renterPhone,
                });
            },
            onError: () => {
                dispatch(
                    showModal({
                        text: t('EditProfile.failedText'),
                        status: 'failed',
                    }),
                );
                setTimeout(() => {
                    dispatch(hideModal());
                }, 2000);
                dispatch(hideProgressIndicator());
            },
        });
    };

    return (
        <ScreenLayout
            testId="EditProfileScreen"
            backgroundColor="light"
            padding={20}
            flex>
            <View>
                {/* <View>
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.xl,
                            color: Color.primaryColor,
                        }}>
                        {t('EditProfile.textName')} :
                    </Text>
                    <TextInput
                        value={userData.name}
                        pointerEvents="none"
                        style={{
                            borderWidth: 1,
                            borderColor: Color.primaryColor,
                            borderRadius: 10,
                            padding: 10,
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.xl,
                            color: Color.primaryColor,
                        }}
                    />
                </View>
                <View>
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.xl,
                            color: Color.primaryColor,
                        }}>
                        {t('EditProfile.textEmail')}
                    </Text>
                    <TextInput
                        value={userData.email}
                        pointerEvents="none"
                        style={{
                            borderWidth: 1,
                            borderColor: Color.primaryColor,
                            borderRadius: 10,
                            padding: 10,
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.xl,
                            color: Color.primaryColor,
                        }}
                    />
                </View> */}
                <View>
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.xl,
                            color: Color.primaryColor,
                        }}>
                        {t('EditProfile.textPhoneNumber')}
                    </Text>
                    <TextInput
                        testID="phoneNumberInput"
                        pointerEvents={userData.phoneNumber ? 'none' : 'auto'}
                        value={
                            userData.phoneNumber
                                ? userData.phoneNumber
                                : renterPhone
                        }
                        onChangeText={setRenterPhone}
                        style={{
                            borderWidth: 1,
                            borderColor: Color.primaryColor,
                            borderRadius: 10,
                            padding: 10,
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.xl,
                            color: Color.primaryColor,
                        }}
                        keyboardType="numeric"
                    />
                </View>
            </View>

            <Button
                testID="buttonSave"
                label={t('EditProfile.buttonSave')}
                backgroundColor={Color.primaryColor}
                borderRadius={10}
                marginT-10
                onPress={handleSave}
            />
        </ScreenLayout>
    );
};

export default EditProfile;
