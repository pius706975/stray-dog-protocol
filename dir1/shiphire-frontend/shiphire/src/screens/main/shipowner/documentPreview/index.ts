export {default as OwnerDocumentPreview} from './OwnerDocumentPreview';
