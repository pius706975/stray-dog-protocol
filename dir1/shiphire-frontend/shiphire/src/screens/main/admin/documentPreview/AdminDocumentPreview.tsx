import React from 'react';
import { useTranslation } from 'react-i18next';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import { Color } from '../../../../configs';
import { modalSlice } from '../../../../slices';
import { AdminDocumentPreviewProps } from '../../../../types';

const AdminDocumentPreview: React.FC<AdminDocumentPreviewProps> = ({
    route,
}) => {
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const { documentUrl, isButtonActive, btnText } = route.params;
    const { t } = useTranslation('common');

    const handleError = () => {
        dispatch(
            showModal({
                status: 'failed',
                text: t('DocumentPreview.failedLoad'),
            }),
        );
        setTimeout(() => {
            dispatch(hideModal());
        }, 3000);
    };

    return (
        <>
            <ScreenLayout testId="DocumentShipScreen" backgroundColor="light">
                {documentUrl && (
                    <PDFView
                        fadeInDuration={250.0}
                        style={{
                            flex: 1,
                            borderWidth: 1,
                            backgroundColor: Color.bgColor,
                        }}
                        resource={documentUrl}
                        resourceType={'url'}
                        onError={error => handleError()}
                    />
                )}
            </ScreenLayout>
            {isButtonActive && (
                <View
                    style={{
                        position: 'absolute',
                        bottom: 30,
                        right: 30,
                    }}>
                    <Button
                        title={btnText ? btnText : ''}
                        color="success"
                        onSubmit={() => {
                            // onClick!!();
                        }}
                    />
                </View>
            )}
        </>
    );
};

export default AdminDocumentPreview;
