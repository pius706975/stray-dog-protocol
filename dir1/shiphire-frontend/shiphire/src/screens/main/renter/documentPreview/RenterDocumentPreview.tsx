import React from 'react';
import { useTranslation } from 'react-i18next';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import { Color } from '../../../../configs';
import { modalSlice } from '../../../../slices';
import { RenterDocumentPreviewProps } from '../../../../types';

const RenterDocumentPreview: React.FC<RenterDocumentPreviewProps> = ({
    route,
}) => {
    const dispatch = useDispatch();
    const { t } = useTranslation('common');
    const { hideModal, showModal } = modalSlice.actions;
    const { documentUrl, isEditable } = route.params;

    const handleError = () => {
        dispatch(
            showModal({
                status: 'failed',
                text: t('RenterDocPreview.failedLoad'),
            }),
        );
        setTimeout(() => {
            dispatch(hideModal());
        }, 3000);
    };

    return (
        <>
            <ScreenLayout testId="DocumentShipScreen" backgroundColor="light">
                {documentUrl && (
                    <PDFView
                        fadeInDuration={250.0}
                        style={{
                            flex: 1,
                            borderWidth: 1,
                            backgroundColor: Color.bgColor,
                        }}
                        resource={documentUrl}
                        resourceType={'url'}
                        onError={error => handleError()}
                    />
                )}
            </ScreenLayout>
            {isEditable && (
                <View
                    style={{
                        position: 'absolute',
                        bottom: 30,
                        right: 30,
                    }}>
                    <Button
                        title="Edit Document"
                        color="success"
                        onSubmit={() => {}}
                    />
                </View>
            )}
        </>
    );
};

export default RenterDocumentPreview;
