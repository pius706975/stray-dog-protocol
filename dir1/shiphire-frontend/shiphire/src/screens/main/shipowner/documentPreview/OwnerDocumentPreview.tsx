import React from 'react';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import { Color } from '../../../../configs';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';
import { AcceptRFQRequest, OwnerDocumentPreviewProps } from '../../../../types';
import { useTranslation } from 'react-i18next';
import { useUpdateTransactionAcceptRFQ } from '../../../../hooks';

const OwnerDocumentPreview: React.FC<OwnerDocumentPreviewProps> = ({
    navigation,
    route,
}) => {
    const { documentUrl, shipId, documentName } = route.params;
    const dispatch = useDispatch();
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const { hideModal, showModal } = modalSlice.actions;
    const mutationUpdateTransactionAcceptRFQ = useUpdateTransactionAcceptRFQ();
    const { t } = useTranslation('common');

    const handleError = () => {
        dispatch(
            showModal({
                status: 'failed',
                text: t('DocumentPreview.failedLoad'),
            }),
        );
        setTimeout(() => {
            dispatch(hideModal());
        }, 3000);
    };

    const handleEditDocument = () => {
        navigation.navigate('EditDocumentForm', {
            documentUrl: documentUrl,
            shipId: shipId!!,
            documentName,
        });
    };

    return (
        <>
            <ScreenLayout testId="DocumentShipScreen" backgroundColor="light">
                {documentUrl && (
                    <PDFView
                        fadeInDuration={250.0}
                        style={{
                            flex: 1,
                            borderWidth: 1,
                            backgroundColor: Color.bgColor,
                        }}
                        resource={documentUrl}
                        resourceType={'url'}
                        onError={error => handleError()}
                    />
                )}
            </ScreenLayout>

            {/* <View
                style={{
                    position: 'absolute',
                    bottom: 30,
                    right: 30,
                }}>
                <Button
                    title="Edit Document"
                    color="success"
                    onSubmit={() => {
                        handleEditDocument();
                    }}
                />
            </View> */}
        </>
    );
};

export default OwnerDocumentPreview;
