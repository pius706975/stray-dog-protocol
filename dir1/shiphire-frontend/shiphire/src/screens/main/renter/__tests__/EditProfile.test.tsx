import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainStackParamList } from '../../../../types';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import PDFView from 'react-native-view-pdf';
import { RenterDocumentPreview } from '../documentPreview';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { EditProfile } from '../editProfile';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { USERDATA } from '../../../../configs';
import { act } from 'react-test-renderer';

const queryClient = new QueryClient();
const MainStack = createNativeStackNavigator<MainStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const mockScreen = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainStack.Navigator>
                    <MainStack.Screen
                        name="EditProfile"
                        component={EditProfile}
                    />
                </MainStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

AsyncStorage.setItem(
    USERDATA,
    JSON.stringify({
        name: 'Muhamad Fauzan',
        email: 'usman@email.com',
        googleId: 'renter@email.com',
        phoneNumber: '0808080',
        password: 'test',
        isVerified: false,
    }),
);

describe('Testing edit profile screen', () => {
    describe('Snapshot Testing', () => {
        it('should render edit profile screen correctly', () => {
            const editProfileScreen = render(mockScreen).toJSON();
            expect(editProfileScreen).toMatchSnapshot();
        });
    });
    describe('Unit Testing', () => {
        it('should show an alert when success add phone number', async () => {
            const { getByTestId } = render(mockScreen);

            mockAdapter
                .onPost('/user/add-phone-number')
                .reply(200, { status: 'success' });

            const buttonSave = getByTestId('buttonSave');

            await act(async () => {
                fireEvent.press(buttonSave);
            });

            waitFor(() => {
                const modalState = store.getState().modal;

                expect(modalState).toEqual({
                    visible: true,
                    status: 'success',
                    text: 'EditProfile.successText',
                });
            });
        });
        it('should show an alert when failed add phone number', async () => {
            jest.useFakeTimers();
            const { getByTestId } = render(mockScreen);

            mockAdapter
                .onPost('/user/add-phone-number')
                .reply(500, { status: 'failed' });

            const buttonSave = getByTestId('buttonSave');

            await act(async () => {
                fireEvent.press(buttonSave);
            });
        });
    });
});
