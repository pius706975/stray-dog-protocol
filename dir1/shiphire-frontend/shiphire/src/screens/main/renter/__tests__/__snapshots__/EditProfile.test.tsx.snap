// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing edit profile screen Snapshot Testing should render edit profile screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="EditProfileScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": 1,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 20,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "color": "#216dab",
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 22,
                          },
                        ]
                      }
                    >
                      EditProfile.textPhoneNumber
                    </Text>
                    <TextInput
                      keyboardType="numeric"
                      onChangeText={[Function]}
                      pointerEvents="auto"
                      style={
                        {
                          "borderColor": "#216dab",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "color": "#216dab",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 22,
                          "padding": 10,
                        }
                      }
                      testID="phoneNumberInput"
                      value=""
                    />
                  </View>
                </View>
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onLayout={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#216dab",
                      "borderRadius": 10,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 10,
                      "minWidth": 90,
                      "opacity": 1,
                      "paddingHorizontal": 20,
                      "paddingVertical": 9.5,
                    }
                  }
                  testID="buttonSave"
                >
                  <Text
                    fsTagName="unmask"
                    numberOfLines={1}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        [
                          {
                            "backgroundColor": "transparent",
                            "flexDirection": "row",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "lineHeight": 24,
                          },
                          {
                            "color": "#FFFFFF",
                          },
                          undefined,
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                    testID="buttonSave.label"
                  >
                    EditProfile.buttonSave
                  </Text>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="EditProfile"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
